src/mbgl/style/expression/compound_expression.cpp

```diff
@@ -42,23 +42,23 @@ template <class R, class... Params>
 struct Signature<R (Params...)> : SignatureBase {
     using Args = std::array<std::unique_ptr<Expression>, sizeof...(Params)>;
     
-    Signature(R (*evaluate_)(Params...)) :
+    Signature(R (*evaluate_)(Params...), const std::string& name) :
         SignatureBase(
             valueTypeToExpressionType<std::decay_t<typename R::Value>>(),
-            std::vector<type::Type> {valueTypeToExpressionType<std::decay_t<Params>>()...}
+            std::vector<type::Type> {valueTypeToExpressionType<std::decay_t<Params>>()...},
+            name
         ),
-        evaluate(evaluate_)
-    {}
+        evaluate(evaluate_)    {}
     
     EvaluationResult apply(const EvaluationContext& evaluationParameters, const Args& args) const {
         return applyImpl(evaluationParameters, args, std::index_sequence_for<Params...>{});
     }
     
-    std::unique_ptr<Expression> makeExpression(const std::string& name,
+    std::unique_ptr<Expression> makeExpression(const std::string& name_,
```

Comment:
I think we can remove the `name` parameter from `makeExpression` now that `Signature`s know their own names.